### ISSUE TITLE
How to use face_recognition in Python so that it can make use of all CPUs ??

### ISSUE THREAD

**Issue by 1warpengine** [ageitgey/face_recognition#443]
* face_recognition version: latest
* Python version: 3.6
* Operating System: Ubuntu

### Description
Running the example I noticed it use only 1 CPU, I'm not talinking about the command line face:recognition but using the lib in python.

**Comment by MLDSBigGuy** [comment]
I think u are asking [this line](https://github.com/ageitgey/face_recognition/blob/master/face_recognition/face_recognition_cli.py#L78)

**Comment by cftang0827** [comment]
It seems like the pool can do face comparing in multiprocessing way. However, the bottleneck of the face recognition computing is face encoding. So is there any way to do face encoding in multi-core way? 

THANKs :)

**Comment by MLDSBigGuy** [comment]
yes. Create a ML model with n_cpus = -1 parameter. Give this model all set of encodings u have. n_cpus = -1 makes use of all the cpus u have to train the encodings.

**Comment by cftang0827** [comment]
Excuse me, do you mean creating a model in dlib from scratch?
Or is there any ways to do transfer learning to modify dlib’s face recognition model?

Thank you for your reply :)

**Comment by cftang0827** [comment]
Actually I think the original issue is about “inference time” but did you mean “traning time”?

If there’s any misunderstanding, please judge me. Thanks

**Comment by MLDSBigGuy** [comment]
I mean this at this step, https://github.com/ageitgey/face_recognition/blob/master/examples/face_recognition_knn.py#L100, add n_jobs = -1 
Ref: http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html
